Orm/Xtensive.Orm/Orm/Linq/Translator.Expressions.cs

```diff
@@ -1594,7 +1594,10 @@ private static ProjectionExpression CreateLocalCollectionProjectionExpression(Ty
     {
       var storedEntityType = translator.State.TypeOfEntityStoredInKey;
       var itemToTupleConverter = ItemToTupleConverter.BuildConverter(itemType, storedEntityType, value, translator.context.Model, sourceExpression);
-      var rsHeader = new RecordSetHeader(itemToTupleConverter.TupleDescriptor, itemToTupleConverter.TupleDescriptor.Select(x => new SystemColumn(translator.context.GetNextColumnAlias(), 0, x)).Cast<Column>());
+      var rsHeader = new RecordSetHeader(
+        itemToTupleConverter.TupleDescriptor,
+        itemToTupleConverter.TupleDescriptor.Select(x => new SystemColumn(translator.context.GetNextColumnAlias(), 0, x)).Cast<Column>().ToArray(itemToTupleConverter.TupleDescriptor.Count)
+      );
```

Comment:
I really think we should use variables like so
```csharp
var tupleDescriptor = itemToTupleConverter.TupleDescriptor;
var columns = tupleDescriptor
  .Select(x => new SystemColumn(translator.context.GetNextColumnAlias(), 0, x))
  .Cast<Column>()
  .ToArray(tupleDescriptor.Count);
var rsHeader = new RecordSetHeader(tupleDescriptor, columns);
```

Reply:
I agree. done